examples/cuda/appl-matmul/main.c

```diff
@@ -188,13 +188,19 @@ int kernel_appl_matmul (int argc, char **argv) {
                 /*****************************************************************************************************************
                  * Copy A & B from host onto device DRAM.
                  ******************************************************************************************************************/
-                void *dst = (void *) ((intptr_t) A_device);
-                void *src = (void *) &A[0];
-                BSG_CUDA_CALL(hb_mc_device_memcpy (&device, dst, src, N * N * sizeof(REAL), HB_MC_MEMCPY_TO_DEVICE)); /* Copy A to the device  */
-
-                dst = (void *) ((intptr_t) B_device);
-                src = (void *) &B[0];
-                BSG_CUDA_CALL(hb_mc_device_memcpy (&device, dst, src, N * N * sizeof(REAL), HB_MC_MEMCPY_TO_DEVICE)); /* Copy B to the device */
+                hb_mc_dma_htod_t htod_A = {
```

Comment:
While this is 'correct' (and much better than using memcpy) you'll save significant simulation time by making a 2-length array of `hb_mc_dma_htod_t` and then having a single call to `hb_mc_device_dma_to_device`.

Reply:
Fixed!